Lib/distutils/tests/test_dist.py

```diff
@@ -354,8 +354,11 @@ def test_classifier_invalid_type(self):
         attrs = {'name': 'Boa', 'version': '3.0',
                  'classifiers': ('Programming Language :: Python :: 3',)}
         msg = "'classifiers' should be a 'list', not 'tuple'"
-        with self.assertRaises(TypeError, msg=msg):
-            Distribution(attrs)
+        with self.assertWarns(RuntimeWarning) as warns:
```

Comment:
Would is better to use `assertWarnsRegex()` here and in other tests. Otherwise `msg` isn't used.